website/project/model.py

```diff
@@ -1417,6 +1420,25 @@ def set_node_license(self, license_id, year, copyright_holders, auth, save=False
         if save:
             self.save()
 
+    def create_keenio_readkey(self):
+        api_readkey = scoped_keys.encrypt(settings.KEEN['public']['master_key'], options={
+            'filters': [{
+                'property_name': 'node.id',
+                'operator': 'eq',
+                'property_value': str(self._id)
+            }],
+            'allowed_operations': ['read']
+        })
+        self.keenio_read_key = api_readkey
+        self.save()
```

Comment:
We did it this way to assign read keys on-demand, as public pages are visited for the first time.  I suppose we could factor it out into a migration, instead.  Would that be preferable?